### PR TITLE
making site ID optional

### PR DESCRIPTION
**Pull request by Baalmart**
**_WHAT DOES THIS PR DO?_**
Makes the site ID optional when querying measurements

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [PLAT-1201]

**_HOW DO I TEST OUT THIS PR?_**
```
cd src/device-registry
npm install
```
`npm run dev-mac` or `npm run dev-pc`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
add events endpoint

[PLAT-1201]: https://airqoteam.atlassian.net/browse/PLAT-1201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/device-registry/routes/api-v1.js [hunk 1]
```diff
@@ -2979,9 +2979,10 @@ router.post(
         .isBoolean()
         .withMessage("is_device_primary should be Boolean"),
       body("*.site_id")
-        .exists()
+        .optional()
```
Comment (Baalmart) [inline src/device-registry/routes/api-v1.js:2982]: in the input validation logic, I have made site_id optional. @NoahNsimbe

File: src/device-registry/routes/api-v2.js [hunk 2]
```diff
@@ -2798,9 +2798,10 @@ router.post(
         .isBoolean()
         .withMessage("is_device_primary should be Boolean"),
       body("*.site_id")
-        .exists()
+        .optional()
```
Comment (Baalmart) [inline src/device-registry/routes/api-v2.js:2801]: Same case for the V2 endpoint, made SITE_ID optional.

**Review by Baalmart — COMMENTED** [review]
@NoahNsimbe , I have made the site_id optional in the validation logic.

I also realise other mandatory fields, what should we do for them? They include:

- `device_id` - MONGO_ID
-  `frequency` - STRING
-` time` - DATETIME

Also, we will need to review why we made site_id mandatory in the first place and then find alternative ways of addressing our target goal. Was it about ensuring that we can retrieve measurements by SITE_ID. I guess there was an obsession with SITEID as some point in time.

**Review by NoahNsimbe — APPROVED** [review]
LGTM. Thanks @Baalmart

**Pull request merged**